src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -193,23 +193,49 @@ public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteO
         {
         }
 
-        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize) :
-            this(ValidateArgsAndOpenPath(path, encoding, bufferSize), encoding, detectEncodingFromByteOrderMarks, bufferSize, leaveOpen: false)
+        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize)
+            : this(ValidateArgsAndOpenPath(path, encoding, bufferSize), encoding, detectEncodingFromByteOrderMarks, bufferSize, leaveOpen: false)
         {
         }
 
+#if NET6_0_OR_GREATER
+        public StreamReader(string path, FileStreamOptions options)
+            : this(path, Encoding.UTF8, true, options)
+        {
+        }
+
+        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, FileStreamOptions options)
+            : this(ValidateArgsAndOpenPath(path, encoding, options), encoding, detectEncodingFromByteOrderMarks, DefaultBufferSize)
+        {
+        }
+
+        private static Stream ValidateArgsAndOpenPath(string path, Encoding encoding, FileStreamOptions options)
+        {
+            ValidateArgs(path, encoding);
+            if (options == null)
+                throw new ArgumentNullException(nameof(options));
+
+            return new FileStream(path, options);
+        }
+#endif
```

Comment:
`StreamReader is not included in [Microsoft.IO.Redist.csproj](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.IO.Redist/src/Microsoft.IO.Redist.csproj), there is no need for the `#if` here

```suggestion
        public StreamReader(string path, FileStreamOptions options)
            : this(path, Encoding.UTF8, true, options)
        {
        }

        public StreamReader(string path, Encoding encoding, bool detectEncodingFromByteOrderMarks, FileStreamOptions options)
            : this(ValidateArgsAndOpenPath(path, encoding, options), encoding, detectEncodingFromByteOrderMarks, DefaultBufferSize)
        {
        }

        private static Stream ValidateArgsAndOpenPath(string path, Encoding encoding, FileStreamOptions options)
        {
            ValidateArgs(path, encoding);
            if (options == null)
                throw new ArgumentNullException(nameof(options));

            return new FileStream(path, options);
        }
```